### DISCUSSION TITLE
Putting the controller at the other end of the controller wire

### DISCUSSION THREAD

**Discussion by Tobbe** [gcormier/megadesk#110] [Ideas]
Hi

I've seen some requests for wifi/bluetooth. And the answer is that there's just not enough room for a wifi chip inside the button housing.

So that got me thinking. The wire from the stock button controller connects to one of the legs. Would it be possible to design megadesk to be installed at that end instead? So plug the "leg end" of the controller wire into megadesk, and then another wire from megadesk into the leg. That would give you infinite room for whatever hardware you wanted. And you could house it all in any generic hardware enclosure.

**Comment by gcormier** [comment]
I had actually started some napkin design on this and called it Megadesk Companion.

Ultimately, I figured the price point for designing and producing a separate board would cost more then buying a simple Arduino-style device and hacking it.

I do have a prototype using the new ESP32-S3-MINI. Unfortunately, last I checked, it was not yet supported by platformio/arduino.

**Reply by Tobbe** [reply]
You say "companion". I take that to mean you're thinking we'd still need today's Megadesk inside the button housing.

Would it be possible to just replace the board inside the button housing with a simple pass-through board, and have all the logic in the "companion"?

![image](https://user-images.githubusercontent.com/30793/187856596-2312fb6f-b19d-4ed3-8dec-a983ab48377b.png)

Or not even a board. Maybe simply just desolder the wires coming into the housing and connect them straight to the ribbon cable as in the sketch above.

**Reply by gcormier** [reply]
Interesting idea and that could certainly work. The discussion on ESP firmware remains relevant. However, this would solve space constraints and allow for off the shelf DCDC converters.
